components/logins/src/sync/engine.rs

```diff
@@ -504,6 +506,22 @@ impl SyncEngine for LoginsSyncEngine {
     }
 }
 
+// This seems misplaced - but everywhere else a mirror row is read we don't need the unknown
+// fields because they are, um, unknown :) So we never merge or inspect them, just send
+// whatever we have in outgoing.
+fn unknown_fields_from_row(
+    row: &rusqlite::Row<'_>,
+) -> Result<Option<super::payload::UnknownFields>> {
```

Comment:
I think I like `Result<UnknownFields>` better -- both for this return type and the argument types for functions like `into_bso()`.   That way there's no ambiguity between `None` and an empty map.  To me, `UnknownFields::new()` communicates "there's no unknown fields" clearer than `None` (maybe we could define `UnknownFields::empty()` to be even more clear)

Reply:
I think this is a little interesting -- should we always assume there is an `unknown_fields` but could be empty? I _slightly_ believe the *absence* of `unknown_fields` (None/null) is more clearer that if that field is null or undefined then we don't need to worry about it. This also keeps the "status quo" of today's world.

However I do understand the argument for an empty map. We'll probably just need to see a potential case where `None` doesn't work and see what ambiguity lies there.

Reply:
Yeah - in this specific case, it's a reflection of a nullable sql column - it's flattened out later, and there's literally zero difference between an empty map and `None` - so I agree that option should die